src/api/client.js

```diff
@@ -0,0 +1,104 @@
+// import axios and create pre-configured version of axios
+import axios from 'axios'
+import { mutate } from 'swr';
+
+const client = axios.create({
+  baseURL: 'http://localhost:3000'
+})
+
+export default client
+
+/**
+ * @typedef PayloadWithId
+ * @type {object}
+ * @property {string} id - The ID of the payload
+ */
+
+/**
+ * Creates an entity associated with the current output
+ * @param {string} endpoint the endpoint that handles the entity
+ * @param {object} payload the payload for the object
+ * @returns {Array} A list containing the response and error respectivly
+ */
+export async function createEntity(endpoint, payload) {
+  let res = null
+  let error = null
+
+  try {
+    res = await client.post(endpoint, payload)
+  } catch (e) {
+    error = e
+  }
+
+  if (!error) {
+    // Dispatch mutate so we know to update the associated endpoint SWR cache
+    mutate(endpoint, (current) => {
+      return [
+        ...current,
+        { id: -1, ...payload }
+      ]
+    })
+  }
+
+  return [res, error]
+}
+
+/**
+ * Update an entity based on a given endpoint
+ * @param {String} endpoint 
+ * @param {PayloadWithId} payload The payload to update
+ * @param {Array}
+ * @returns 
+ */
+export async function updateEntity(endpoint, payload) {
```

Comment:
Same as createEntity, should we allow a third parameter to allow client code to config the axios request?